RecoBTag/SoftLepton/plugins/SoftPFElectronTagInfoProducer.cc

```diff
@@ -42,104 +44,99 @@ SoftPFElectronTagInfoProducer::~SoftPFElectronTagInfoProducer()
-  if(goodvertex){	
-  	for(reco::PFCandidateCollection::const_iterator lepton = leptons.begin(); lepton != leptons.end(); ++lepton) {
-    	reco::TrackBaseRef trkRef(lepton->gsfTrackRef());
-    	reco::SoftLeptonProperties properties=fillElecProperties(*lepton, *jet);
-    	info.insert(trkRef, properties );
-		}	
-	}
-  return info;
-}
-
-reco::SoftLeptonProperties SoftPFElectronTagInfoProducer::fillElecProperties(const reco::PFCandidate &elec, const reco::Jet &jet) {
-  reco::SoftLeptonProperties prop;
-  reco::TransientTrack transientTrack=transientTrackBuilder->build(elec.gsfTrackRef().get());
-  prop.sip2d    = IPTools::signedTransverseImpactParameter(transientTrack, GlobalVector(jet.px(), jet.py(), jet.pz()), *vertex).second.significance();
-  prop.sip3d    = IPTools::signedImpactParameter3D(transientTrack, GlobalVector(jet.px(), jet.py(), jet.pz()), *vertex).second.significance();
-  prop.deltaR   = deltaR(jet, elec);
-  prop.ptRel    = ( (jet.p4().Vect()-elec.gsfElectronRef().get()->p4().Vect()).Cross(elec.gsfElectronRef().get()->p4().Vect()) ).R() / jet.p4().Vect().R(); // | (Pj-Pu) X Pu | / | Pj |
-  float mag = elec.gsfElectronRef().get()->p4().Vect().R()*jet.p4().Vect().R();
-  float dot = elec.gsfElectronRef().get()->p4().Dot(jet.p4());
-  prop.etaRel   = -log((mag - dot)/(mag + dot)) / 2.;
-  prop.ratio    = elec.gsfElectronRef().get()->p() / jet.energy();
-  prop.ratioRel = elec.gsfElectronRef().get()->p4().Dot(jet.p4()) / jet.p4().Vect().Mag2();
-  return prop;
+  	edm::Handle<edm::View<reco::Jet> > theJetCollection;
+  	iEvent.getByToken(token_jets, theJetCollection);
+
+	edm::Handle<edm::View<reco::GsfElectron > > theGEDGsfElectronCollection;
+  	iEvent.getByToken(token_elec, theGEDGsfElectronCollection);
+
+	edm::Handle<reco::BeamSpot> bsHandle;
+  	iEvent.getByToken(token_BeamSpot, bsHandle);
+  	const reco::BeamSpot &beamspot = *bsHandle.product();
+
+	 for (unsigned int i = 0; i < theJetCollection->size(); i++){
+    		edm::RefToBase<reco::Jet> jetRef = theJetCollection->refAt(i);
+		reco::CandSoftLeptonTagInfo tagInfo;
+                tagInfo.setJetRef(jetRef);
+    		std::vector<const reco::GsfElectron *> Elec;
+		for(unsigned int ie=0, ne=theGEDGsfElectronCollection->size(); ie<ne; ++ie){
+			//Get the edm::Ptr and the GsfElectron
+			edm::Ptr<reco::Candidate> lepPtr=theGEDGsfElectronCollection->ptrAt(ie);
+			const reco::GsfElectron* recoelectron=theGEDGsfElectronCollection->refAt(ie).get();
+			const pat::Electron* patelec=dynamic_cast<const pat::Electron*>(recoelectron);
+			if(patelec){
+				if(patelec->passConversionVeto())continue;
+			}
+			else{
+				if(ConversionTools::hasMatchedConversion(*(recoelectron),hConversions,beamspot.position()))continue;
+			}
+			float deta=fabs(recoelectron->eta()-jetRef->eta());
+			float dphi=fabs(recoelectron->phi()-jetRef->phi());
+			float dphi2=dphi<3.14159?dphi:6.2831-dphi;
+			//Make sure that the electron is inside the jet
+			if(sqrt(pow(deta,2)+pow(dphi2,2))>0.5)continue;
```

Comment:
On Feb 4, 2015, at 12:46 PM, Slava Krutelyov <notifications@github.com>
 wrote:

> In RecoBTag/SoftLepton/plugins/SoftPFElectronTagInfoProducer.cc:
> 
> > +		for(unsigned int ie=0, ne=theGEDGsfElectronCollection->size(); ie<ne; ++ie){
> > +			//Get the edm::Ptr and the GsfElectron
> > +			edm::Ptr<reco::Candidate> lepPtr=theGEDGsfElectronCollection->ptrAt(ie);
> > +			const reco::GsfElectron* recoelectron=theGEDGsfElectronCollection->refAt(ie).get();
> > +			const pat::Electron* patelec=dynamic_cast<const pat::Electron*>(recoelectron);
> > +			if(patelec){
> > +				if(patelec->passConversionVeto())continue;
> > +			}
> > +			else{
> > +				if(ConversionTools::hasMatchedConversion(*(recoelectron),hConversions,beamspot.position()))continue;
> > +			}
> > +			float deta=fabs(recoelectron->eta()-jetRef->eta());
> > +			float dphi=fabs(recoelectron->phi()-jetRef->phi());
> > +			float dphi2=dphi<3.14159?dphi:6.2831-dphi;
> > +			//Make sure that the electron is inside the jet
> > +			if(sqrt(pow(deta,2)+pow(dphi2,2))>0.5)continue;
> 
> yes.
> it's somewhat minor, but still
> 

but even better deta*deta and dphi2*dphi2 instead of using pow please…


> —
> Reply to this email directly or view it on GitHub.
>